Makefile

```diff
@@ -141,97 +134,15 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
-MODULE_CREDENTIALS_FLAG=-c oauth2accesstoken:$(GCP_ACCESS_TOKEN)
-else
-MODULE_CREDENTIALS_FLAG=--insecure
-endif
-
-.PHONY: run-with-lm
-run-with-lm: ## Create a k3d cluster and deploy module with the lifecycle-manager. Manager image and module OCI image are pushed to local k3d registry
-run-with-lm: \
-	create-k3d \
-	local-manager-image \
-	create-local-module \
-	fix-module-template \
-	apply-local-template-label \
-	deploy-kyma \
-	deploy-module-template \
-	enable-module \
-	verify-telemetry \
-	verify-kyma \
-
-.PHONY: create-k3d
-create-k3d: kyma ## Create a k3d cluster using Kyma cli .
-	$(KYMA) provision k3d --registry-port ${REGISTRY_PORT} --name ${CLUSTER_NAME} --ci
-
-.PHONY: local-manager-image ## Build and push manager image to local k3d registry
-local-manager-image:
-	@make manager-image \
-		IMG=localhost:${REGISTRY_PORT}/${MODULE_NAME}-manager
-
-.PHONY: create-local-module
-create-local-module:
-	@make create-module \
-		IMG=k3d-${REGISTRY_NAME}:${REGISTRY_PORT}/${MODULE_NAME}-manager \
-		MODULE_REGISTRY=localhost:${REGISTRY_PORT}
-
-.PHONY: create-module
-create-module: kyma kustomize ## Build the module and push it to a registry defined in MODULE_REGISTRY.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
-	$(KYMA) alpha create module --name kyma-project.io/module/${MODULE_NAME} --version ${MODULE_VERSION} --channel ${MODULE_CHANNEL} --default-cr ./config/samples/operator_v1alpha1_telemetry.yaml --registry ${MODULE_REGISTRY} ${MODULE_CREDENTIALS_FLAG}
-
-.PHONY: fix-module-template
-fix-module-template: ## Create template-k3d.yaml based on template.yaml with right URLs.
-	@cat template.yaml \
-	| sed -e 's/${REGISTRY_PORT}/5000/g' \
-		  -e 's/localhost/k3d-${REGISTRY_NAME}.localhost/g' \
-		> template-k3d.yaml
-
-.PHONY: apply-local-template-label
-apply-local-template-label: ## Apply a marker label to be read by the lifecycle manager.
-	kubectl label --local=true -f ./template-k3d.yaml operator.kyma-project.io/use-local-template=true -oyaml > template-k3d-with-label.yaml
-
-.PHONY: deploy-kyma
-deploy-kyma: kyma ## Deploy kyma which includes the deployment of the lifecycle-manager.
-	$(KYMA) alpha deploy \
-		--ci \
-		--force-conflicts
-
-.PHONY: deploy-module-template
-deploy-module-template: ## Deploy the ModuleTemplate in the cluster.
-	kubectl apply -f template-k3d-with-label.yaml
-
-.PHONY: enable-module
-enable-module: kyma ## Enable the module.
-	$(KYMA) alpha enable module ${MODULE_NAME} --channel ${MODULE_CHANNEL}
-
-.PHONY: verify-telemetry
-verify-telemetry: ## Wait for Telemetry CR to be in Ready state.
-	@hack/verify_telemetry_status.sh
-
-.PHONY: verify-kyma
-verify-kyma: ## Wait for Kyma CR to be in Ready state.
-	@hack/verify_kyma_status.sh
+.PHONY: deploy-with-lifecycle-manager
+deploy-with-lifecycle-manager: kyma kustomize ## Create a k3d cluster and deploy module with the lifecycle manager. Manager image and module image are pushed to local k3d registry
```

Comment:
I think the naming of the make targets is a bit inconsistent. We have the `deploy` target that deploys the telemetry-manager to an existing cluster. This target rather represents an e2e-test for the Kyma module and includes provisioning of a k3d cluster, validating, etc.. I suggest to align the naming and the scope with the `e2e-test` target. Since the `e2e-test` also deletes the k3d cluster, let's do the same here. Also align the `##@` comments (they are used when calling `make  help`)

Reply:
The target is renamed to `e2e-deploy-module`. After discussion, we decided not to delete the k3d cluster as it will be useful for development purposes as well and not only testing. For example, if we make some changes and want to deploy the module with the lifecycle-manager to check something, then we can just execute the make target to have everything setup in a local k3d cluster